lib/remoteproc/remoteproc_virtio.c

```diff
@@ -40,6 +41,11 @@ static unsigned char rproc_virtio_get_status(struct virtio_device *vdev)
 	rpvdev = metal_container_of(vdev, struct remoteproc_virtio, vdev);
 	vdev_rsc = rpvdev->vdev_rsc;
 	io = rpvdev->vdev_rsc_io;
+
+#ifdef VIRTIO_CACHED_RSC_TABLE
```

Comment:
Can we move #ifdef part to header file ?
May be we can introduce function to check if cache is on or not and then based on that we can decide if we need to call metal_cache_invalidate or not.

Reply:
So far, in OpenAMP there are 2 ways of doing cache invalidation:
-  rpmsg: https://github.com/OpenAMP/open-amp/blob/main/lib/rpmsg/rpmsg_virtio.c#L138
- virtqueue: https://github.com/OpenAMP/open-amp/blob/main/lib/virtio/virtqueue.c#L32

I've chose the first one.
You're suggesting the second one - just to put the defines in a header, not in .c file, right?

Reply:
@carlocaione yes, will do like this.

Reply:
> @carlocaione yes, will do like this.

bonus point if you can clean up these two ways of doing the same thing :stuck_out_tongue_winking_eye: (in a different PR)

Reply:
Thanks. Yes second way works.